packages/react/src/use-object.ui.test.tsx

```diff
@@ -69,208 +73,180 @@ describe('text stream', () => {
     beforeEach(() => {
       render(<TestComponent />);
     });
-    describeWithTestServer(
-      "when the API returns 'Hello, world!'",
-      {
-        url: '/api/use-object',
-        type: 'stream-values',
-        content: ['{ ', '"content": "Hello, ', 'world', '!"'],
-      },
-      ({ call }) => {
-        beforeEach(async () => {
-          await userEvent.click(screen.getByTestId('submit-button'));
+
+    describe("when the API returns 'Hello, world!'", () => {
+      beforeEach(async () => {
+        server.urls['/api/use-object'].response = {
+          type: 'stream-chunks',
+          chunks: ['{ ', '"content": "Hello, ', 'world', '!"'],
+        };
+        await userEvent.click(screen.getByTestId('submit-button'));
+      });
+
+      it('should render stream', async () => {
+        await screen.findByTestId('object');
+        expect(screen.getByTestId('object')).toHaveTextContent(
+          JSON.stringify({ content: 'Hello, world!' }),
+        );
+      });
+
+      it("should send 'test' to the API", async () => {
+        expect(await server.calls[0].requestBody).toBe('test-input');
+      });
+
+      it('should not have an error', async () => {
+        await screen.findByTestId('error');
+        expect(screen.getByTestId('error')).toBeEmptyDOMElement();
+        expect(onErrorResult).toBeUndefined();
+      });
+    });
+
+    describe('isLoading', () => {
+      it('should be true while loading', async () => {
+        const controller = new TestResponseController();
+        server.urls['/api/use-object'].response = {
+          type: 'controlled-stream',
+          controller,
+        };
+
+        controller.write('{"content": ');
+        await userEvent.click(screen.getByTestId('submit-button'));
+
+        // wait for element "loading" to have text content "true":
+        await waitFor(() => {
+          expect(screen.getByTestId('loading')).toHaveTextContent('true');
         });
 
-        it('should render stream', async () => {
-          await screen.findByTestId('object');
+        controller.write('"Hello, world!"}');
+        controller.close();
+
+        // wait for element "loading" to have text content "false":
+        await waitFor(() => {
+          expect(screen.getByTestId('loading')).toHaveTextContent('false');
+        });
+      });
+    });
+
+    describe('stop', () => {
```

Comment:
is the separate `describe` needed here? if it's just a single `it` we can remove the `describe` also above)

Reply:
i'd use semantic diff, githubs diffs are wrong https://app.semanticdiff.com/gh/vercel/ai/pull/5509/changes#packages/react/src/use-object.ui.test.tsx?ignore_comments=false

Reply:
oh wait i see what you mean, yeah we can